packages/eas-cli/src/credentials/android/actions/SetupGoogleServiceAccountKey.ts

```diff
@@ -0,0 +1,80 @@
+import nullthrows from 'nullthrows';
+
+import {
+  CommonAndroidAppCredentialsFragment,
+  GoogleServiceAccountKeyFragment,
+} from '../../../graphql/generated';
+import Log from '../../../log';
+import { promptAsync } from '../../../prompts';
+import { Context } from '../../context';
+import { MissingCredentialsNonInteractiveError } from '../../errors';
+import { AppLookupParams } from '../api/GraphqlClient';
+import { AssignGoogleServiceAccountKey } from './AssignGoogleServiceAccountKey';
+import { CreateGoogleServiceAccountKey } from './CreateGoogleServiceAccountKey';
+import { UseExistingGoogleServiceAccountKey } from './UseExistingGoogleServiceAccountKey';
+
+export class SetupGoogleServiceAccountKey {
+  constructor(private app: AppLookupParams) {}
+
+  private async isGoogleServiceAccountKeySetupAsync(ctx: Context): Promise<boolean> {
+    const appCredentials = await ctx.android.getAndroidAppCredentialsWithCommonFieldsAsync(
+      this.app
+    );
+    return !!appCredentials?.googleServiceAccountKeyForSubmissions;
+  }
```

Comment:
This is just me, but I prefer to place private methods below public ones. If I read the code of a class I first look at public methods and usually I don't need to dig into private ones.